content/en/apps/guides/hosting/core-developer.md

```diff
@@ -0,0 +1,240 @@
+
+To deploy the CHT in production, see either [AWS hosting]({{< relref "apps/guides/hosting/self-hosting.md" >}}) or [Self hosting]({{< relref "apps/guides/hosting/ec2-setup-guide.md" >}}){{% /alert %}}
+
+
+## The Happy Path Installation
+
+This CHT Core developer guide will have you install NodeJS, npm, Grunt and CouchDB (via Docker) on your local workstation. These instructions should work verbatim on Ubuntu 18-22, but will need tweaks for MacOS (via `brew`) or Windows (via WSL2).
+
+### Install NodeJS, npm, grunt and Docker
+
+First, update your current Ubuntu packages and install some supporting tools via `apt`:
+
+```shell
+sudo apt update && sudo apt -y dist-upgrade
+sudo apt -y install xsltproc curl uidmap jq
+```
+
+Then install `nvm`, add it to your path and install NodeJS 12:
+
+```shell
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | $0
+. ~/.$0rc
+nvm install 12
+```
+
+Now let's ensure NodeJS 12 and npm 6 were installed. This should output version 12.x.x for NodeJS and 6.x.x for `npm`:
+
+```shell
+node -v && npm -v
+```
+
+With NodeJS out of the way, let's install `grunt` via `npm`:
+
+```shell
+npm install -g grunt-cli
+```
+
+Install Docker:
+
+```shell
+curl -fsSL get.docker.com -o get-docker.sh && sh get-docker.sh
+```
+
+It's easier if you don't always have to run `sudo` for all your Docker calls, so let's set that up:
+
+```shell
+dockerd-rootless-setuptool.sh install
+echo "export PATH=/usr/bin:$PATH" >> ~/.$0rc
+echo "export DOCKER_HOST=unix:///run/user/1000/docker.sock" >> ~/.$0rc
+. ~/.$0rc
+```
+
+In order for Docker to boot correctly, restart entire machine, which will complete the "Install" Section:
+
+```shell
+sudo reboot
+```
+
+### CouchDB Setup in Docker
+
+Before we get started, let's run the simple `hello-world` Docker container. This will ensure docker is working as expected and output "Hello from Docker!" as well as some other intro text:
+
+```shell
+docker run hello-world
+```
+
+Now that we know Docker is set up, let's start our CouchDB container. Note this will run in the background and store its data in `/home/YOUR-USER/cht-docker`. The login for your CHT instance will be `medic` and the password will be `password`:
+
+```shell
+docker run -d -p 5984:5984 -p 5986:5986 --name medic-couchdb -e COUCHDB_USER=medic -e COUCHDB_PASSWORD=password -v ~/cht-docker/local.d:/opt/couchdb/data -v ~/cht-docker/local.d:/opt/couchdb/etc/local.d apache/couchdb:2
+```
+
+Let's ensure CouchDB is set up with a test `curl` call. This should show "nonode@nohost" in JSON:
+
+```shell
+curl -X GET "http://medic:password@localhost:5984/_membership" | jq
+```
+
+Every time you run any `grunt` or `node` commands, it will expect `COUCH_NODE_NAME` and `COUCH_URL` environment variables to be set:
+
+```shell
+echo "export COUCH_NODE_NAME=nonode@nohost">> ~/.$0rc
+echo "export COUCH_URL=http://medic:password@localhost:5984/medic">> ~/.$0rc
+. ~/.$0rc
+echo $COUCH_NODE_NAME && echo $COUCH_URL
```

Comment:
Maybe split this out as a separate block so we can say "to make sure it's working, run this:"

Reply:
yup yup! done in 39f1bbb